clouddriver-openstack/src/main/java/org/openstack4j/model/image/v2/Image.java

```diff
@@ -0,0 +1,267 @@
+ * not yet been released. After [this commit](https://github.com/ContainX/openstack4j/commit/bb78146aa914b855c2d77e2b4d42e455190fb8eb)
+ * makes it into a proper release, this file and org.openstack4j.openstack.image.v2.domain.GlanceImage should be removed, and the
+ * spinnaker dependencies updated to include that new openstack4j release.
+ */
+
+/**
+ * A Glance v2.0-2.3 Image
+ * @author emjburns
+ * @see http://developer.openstack.org/api-ref-image-v2.html#showImage-v2
+ */
+public interface Image extends BasicResource, Buildable<ImageBuilder> {
+
+  public enum ImageStatus {
+    /**
+     * Image status is not one of the documented options.
+     * http://docs.openstack.org/developer/glance/statuses.html
+     */
+    UNRECOGNIZED,
+    /**
+     * The image identifier has been reserved for an image in the Glance registry.
+     * No image data has been uploaded to Glance and the image size
+     * was not explicitly set to zero on creation.
+     */
+    QUEUED,
+    /**
+     * Denotes that an image’s raw data is currently being uploaded to Glance.
+     * When an image is registered with a call to POST /images and there is an
+     * x-image-meta-location header present, that image will never be in the saving status
+     * (as the image data is already available in some other location).
+     */
+    SAVING,
+    /**
+     * Denotes an image that is fully available in Glance. This occurs when the
+     * image data is uploaded, or the image size is explicitly set to zero on creation.
+     */
+    ACTIVE,
+    /**
+     * Denotes that access to image data is not allowed to any non-admin user.
+     * Prohibiting downloads of an image also prohibits operations like image
+     * export and image cloning that may require image data.
+     */
+    DEACTIVATED,
+    /**
+     * Denotes that an error occurred during the uploading of an image’s data,
+     * and that the image is not readable.
+     */
+    KILLED,
+    /**
+     * Glance has retained the information about the image, but it is no longer
+     * available to use. An image in this state will be removed automatically at a later date.
+     */
+    DELETED,
+    /**
+     * This is similar to deleted, however, Glance has not yet removed
+     * the image data. An image in this state is not recoverable.
+     */
+    PENDING_DELETE;
+
+    @JsonCreator
+    public static ImageStatus value(String v)
+    {
+      if (v == null) return UNRECOGNIZED;
+      try {
+        return valueOf(v.toUpperCase());
+      } catch (IllegalArgumentException e) {
+        return UNRECOGNIZED;
+      }
+    }
+
+    @JsonValue
+    public String value() {
+      return name().toLowerCase();
+    }
+  }
+
+  public enum ImageVisibility {
+    PUBLIC,
+    PRIVATE,
+    UNKNOWN;
+
+    @JsonCreator
+    public static ImageVisibility forValue(String value) {
+      if (value != null)
```

Comment:
Nit: prefer the opening bracket on the same line as the `if`. Not sure if this is the convention for the openstack provider, but it is not in the other providers.